src/vmecpp/__init__.py

```diff
@@ -282,7 +296,12 @@ def _to_cpp_vmecindatapywrapper(self) -> _vmecpp.VmecINDATAPyWrapper:
         cpp_indata._set_mpol_ntor(self.mpol, self.ntor)
         for attr in readonly_attrs - {"mpol", "ntor"}:
             # now we can set the elements of the readonly_attrs
-            getattr(cpp_indata, attr)[:] = getattr(self, attr)
+            value = getattr(self, attr)
+
+            # Asymmetric fields are only populated when lasym==True
+            # so we need to skip them for itemwise assignment
+            if value is not None:
+                getattr(cpp_indata, attr)[:] = value
```

Comment:
this can end up silently doing weird things in case of bugs elsewhere in the code. let's assert that the preconditions we expect are satisfied:

```suggestion
            if value is None:  # must be an asymmetric field and lasym == True
                assert attr in {"rbs", "zbc", "zaxis_c", "raxis_s"} and not cpp_indata.lasym
            else:
                getattr(cpp_indata, attr)[:] = value
```